src/uu/ls/src/ls.rs

```diff
@@ -1971,10 +2075,43 @@ fn display_item_long(
             }
         }
 
+        #[cfg(unix)]
+        let leading_char = {
+            if let Some(Some(ft)) = item.ft.get() {
+                if ft.is_char_device() {
+                    "c"
+                } else if ft.is_block_device() {
+                    "b"
+                } else if ft.is_symlink() {
+                    "l"
+                } else if ft.is_dir() {
+                    "d"
+                } else {
+                    "-"
+                }
+            } else {
+                "-"
+            }
+        };
+        #[cfg(not(unix))]
+        let leading_char = {
+            if item.ft.get().is_some() && item.ft.get().unwrap().is_some() {
```

Comment:
This expression can be cleaned up like this other one

Reply:
Done.